rfc/rfc006-distributed-registry.md

```diff
@@ -0,0 +1,331 @@
+depending on what was lost:
+
+1. If the vendor still has a valid signing certificate it COULD sign the update using that private key.
+2. Otherwise, when the signing certificate has expired or the private key is lost as well, the vendor MUST also provide
+new proof to authenticate ownership of the vendor. The new proof MUST be the first operation in the **patch** field.
+
+Example update:
+```json
+{
+  "patch": [
+    { "op": "add", "path": "/prfs", "value": "some new proof" },
+    { "op": "replace", "path": "/crts", "value": ["CA certificate as base64 ASN.1"] }
+  ]
+}
+``` 
+
+##### CA certificate private key leakage
+In case the vendor leaks its CA private key (or it gets stolen), it MUST remove the associated certificate as soon as possible
+and register a new one.
+In the same update it MUST register new proof and CA certificate as in the key loss process.
+Example update:
+```json
+{
+  "patch": [
+    { "op": "add", "path": "/prfs", "value": "some new proof" },
+    { "op": "replace", "path": "/crts", "value": ["CA certificate as base64 ASN.1"] }
+  ]
+}
+``` 
+
+To avoid attackers re-adding the previously removed CA certificate, the application SHOULD allow administrators to
+ register compromised X.509 certificate thumbprints with the date it was compromised. The application SHALL NOT accept
+ certificates or signatures on or after this date. Communication of network participants regarding this compromised
+ certificate list SHOULD happen out-of-band over an authenticated channel.
+
+### 5.2. Organization
+This document registers a care organization as a client of registered vendor (section 5.1). It's identified by the type `registry/organization`.
+
+#### Fields
+The following fields MUST be present:
+* **id**: MUST contain the organization's ID encoded as URN. This can be any valid URN, for instance an AGB-code or
+  KVK-number e.g.: `urn:oid:2.16.840.1.113883.2.4.6.1:06123456` (AGB-code). It is used to refer to the organization by the
+  vendor itself or other vendors. The organisation ID SHALL NOT be changed.
+* **vid** (vendor ID): MUST be the organization's software vendor's ID as URN (e.g. `urn:oid:1.3.6.1.4.1.54851.4:1234`).
+  The vendor ID SHALL NOT be changed; relation to a care organization SHALL NOT be transfered to another vendor. 
+* **name**: MUST contain the organization's commonly known name as string. Only alphanumeric, dashes (`-`) and space characters are allowed,
+  it SHALL NOT be empty or only contain spaces. It MUST conform to the following regex: `[a-zA-Z0-9 ]+`.
+* **svcs** (services): MUST contain an array of services this organization offers (see "Services" below).
+* **prfs** (proofs): MUST contain an array with cryptographic proofs asserting the organization as being an actual care organization.
+  at the same it serves as commitment from the care organization that it uses the vendor's (identified by **vid**) software.
+  See section 7 for supported proof types.
+
+##### Services
```

Comment:
Maybe services need to be a different object type. A service can be turned on and off per organization. This would be done independently from the organization registration. A question that also arises: must a organization sign something when a service is enabled? In the current workflow, the vendor handles this for the organization.

Services also require a nbf

Reply:
After discussion; Service will be modeled as separate object. Since we currently don't have practical authentication means to use as organization proof, proofs will be added in a later version.